nexus/src/config.rs

```diff
@@ -124,22 +122,15 @@ impl Default for Tunables {
 
 /// Configuration for a nexus server
 #[derive(Clone, Debug, Deserialize, PartialEq, Serialize)]
-pub struct Config {
-    /// Dropshot configuration for external API server
-    pub dropshot_external: ConfigDropshot,
-    /// Dropshot configuration for internal API server
-    pub dropshot_internal: ConfigDropshot,
-    /// Identifier for this instance of Nexus
-    pub id: uuid::Uuid,
+pub struct PackageConfig {
     /// Console-related tunables
     pub console: ConsoleConfig,
     /// Server-wide logging configuration.
     pub log: ConfigLogging,
-    /// Database parameters
-    pub database: db::Config,
     /// Authentication-related configuration
     pub authn: AuthnConfig,
     /// Timeseries database configuration.
+    // TODO: Should this be removed? Nexus needs to initialize it.
```

Comment:
This should probably be a runtime configuration setting, yes. I'm not sure whether that's nexus or RSS that'd be doing that. On the one hand, it might be nice to have some telemetry from early in the setup process. On the other, as it stands, `nexus` needs to be up (and CRDB) for `oximeter` to be collecting data from any clients. It's possible that some other component (not `oximeter`, maybe RSS?) could be writing data into ClickHouse during this phase, but I don't know how comfortable I am with that in general.

TL;DR, yes, but not sure who does that, probably Nexus.

Reply:
Currently:

- Oximeter is launched by RSS, determined by hard-coded RSS configuration data

In the future:

- I hope Nexus manages Oximeter instances (provisioning them, making sure one is always up)
- ... and at that point, this argument probably goes away, because RSS shouldn't be doing the allocation anymore

Reply:
FYI this gets resolved in https://github.com/oxidecomputer/omicron/pull/1240 (Nexus can infer the name by DNS) and further in https://github.com/oxidecomputer/omicron/pull/1241 + https://github.com/oxidecomputer/omicron/pull/1216 , where Nexus provision oximeter / clickhouse wherever it wants.